segmentation_labeling_app/rois/rois.py

```diff
@@ -0,0 +1,215 @@
+are used to define drawing parameters as well as other ROI class
+methods useful for post processing required to display to end user.
+"""
+from typing import List, Tuple, Union
+
+from scipy.sparse import coo_matrix
+import scipy.ndimage
+import numpy as np
+import cv2
+
+import segmentation_labeling_app.utils.query_utils as query_utils
+from segmentation_labeling_app.utils.math_utils import get_magnitude
+
+stroke_weight = 1.125
+
+
+class ROI:
+
+    def __init__(self,
+                 coo_rows: Union[np.array, List[int]],
+                 coo_cols: Union[np.array, List[int]],
+                 coo_data: Union[np.array, List[float]],
+                 image_shape: Tuple[int, int],
+                 segmentation_id: int,
+                 roi_id: int):
+        self.coo_rows = coo_rows
+        self.coo_cols = coo_cols
+        self.coo_data = coo_data
+        self.image_shape = image_shape
+        self.segmentation_id = segmentation_id
+        self.roi_id = roi_id
+        self._sparse_coo = None
+        self._dense_mat = None
+        self._edges = None
+
+    @classmethod
+    def roi_from_query(cls, segmentation_id:int,
+                       roi_id: int) -> "ROI":
+        """
+        Queries and builds ROI object by querying LIMS table for
+        produced labeling ROIs.
+        Args:
+            segmentation_id: Id of the segmentation run
+            roi_id: Unique Id of the ROI to be loaded
+
+        Returns: ROI object for the given segmentation_id and roi_id
+        """
+        label_vars = query_utils.get_labeling_env_vars()
+
+        shape = query_utils.query(
+            f"SELECT * FROM public.segmentation_runs WHERE id={segmentation_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)[0]['video_shape']
+
+        roi = query_utils.query(
+            f"SELECT * FROM public.rois WHERE "
+            f"segmentation_run_id={segmentation_id} AND id={roi_id}",
+            user=label_vars.user,
+            host=label_vars.host,
+            database=label_vars.database,
+            port=label_vars.port,
+            password=label_vars.password)
+        return ROI(coo_rows=roi['coo_rows'][0],
+                   coo_cols=roi['coo_cols'][0],
+                   coo_data=roi['coo_data'][0],
+                   image_shape=shape,
+                   segmentation_id=segmentation_id,
+                   roi_id=roi_id)
+
+    @property
+    def sparse_coo(self):
+        """
+        Produces a sparse coo matrix scipy object from the
+        Returns: A sparse_csc scipy object for the data
+        """
+        if self._sparse_coo is None:
+            self._sparse_coo = coo_matrix((
+                self.coo_data,
+                (self.coo_rows,
+                 self.coo_cols)),
+                shape=self.image_shape)
+        return self._sparse_coo
+
+    @property
```

Comment:
Do not need this? If the user wants this, they can just `roi._sparse_coo.toarray()`